contrib/xds/test/kv_store_xds_delegate_integration_test.cc

```diff
@@ -0,0 +1,337 @@
+#include "envoy/admin/v3/config_dump.pb.h"
+#include "envoy/service/runtime/v3/rtds.pb.h"
+#include "envoy/service/secret/v3/sds.pb.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/http_integration.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+static constexpr char SDS_CLUSTER_NAME[] = "sds_cluster.lyft.com";
+static constexpr char RTDS_CLUSTER_NAME[] = "rtds_cluster";
+static constexpr char CLIENT_CERT_NAME[] = "client_cert";
+
+std::string bootstrapConfig() {
```

Comment:
I do think this is worth unit tests as well, as they're easier for testing updates, eviction etc.